wetterdienst/indexing/file_index_creation.py

```diff
@@ -1,69 +1,120 @@
-    """
-    parameter_path = build_path_to_parameter(parameter, time_resolution, period_type)
-
-    files_server = list_files_of_climate_observations(parameter_path, recursive=True)
 
-    files_server = pd.DataFrame(
-        files_server, columns=[DWDMetaColumns.FILENAME.value], dtype="str"
+    file_index = _create_file_index_for_dwd_server(
+        parameter,
+        time_resolution,
+        period_type,
+        DWDCDCDataPath.CLIMATE_OBSERVATIONS
     )
 
-    # Filter for .zip files
-    files_server = files_server[files_server.FILENAME.str.endswith(ARCHIVE_FORMAT)]
+    file_index = file_index[file_index[DWDMetaColumns.FILENAME.value].str.startswith(ArchiveFormat.ZIP.value)]
 
-    files_server[DWDMetaColumns.FILENAME.value] = files_server[
-        DWDMetaColumns.FILENAME.value
-    ].str.replace(f"{DWD_CDC_PATH}/{DWD_CLIM_OBS_GERMANY_PATH}/", "")
-
-    files_server[DWDMetaColumns.STATION_ID.value] = files_server[
+    file_index[DWDMetaColumns.STATION_ID.value] = file_index[
         DWDMetaColumns.FILENAME.value
     ].apply(lambda x: re.findall(STATION_ID_REGEX, x)[0])
 
-    files_server.loc[:, DWDMetaColumns.STATION_ID.value] = files_server.loc[
+    file_index.loc[:, DWDMetaColumns.STATION_ID.value] = file_index.loc[
         :, DWDMetaColumns.STATION_ID.value
     ].astype(int)
 
-    files_server = files_server.sort_values(
+    file_index = file_index.sort_values(
         by=[DWDMetaColumns.STATION_ID.value, DWDMetaColumns.FILENAME.value]
     )
 
-    return files_server.loc[
+    return file_index.loc[
         :, [DWDMetaColumns.STATION_ID.value, DWDMetaColumns.FILENAME.value]
     ]
 
 
+@lru_cache
+def create_file_index_for_radolan(
+        time_resolution: TimeResolution
+) -> pd.DataFrame:
+    file_index = pd.DataFrame()
+
+    for period_type, radolan_dt_regex, radolan_dt_format in zip((PeriodType.HISTORICAL, PeriodType.RECENT), (RADOLAN_HISTORICAL_DT_REGEX, RADOLAN_RECENT_DT_REGEX), (DatetimeFormat.YM.value, DatetimeFormat.ymdhm.value)):
+        file_index_period = _create_file_index_for_dwd_server(
+            Parameter.RADOLAN,
+            time_resolution,
+            period_type,
+            DWDCDCDataPath.GRIDS_GERMANY
+        )
+
+        file_index_period = file_index_period[file_index_period[DWDMetaColumns.FILENAME.value].str.endswith(
+            (ArchiveFormat.GZ.value, ArchiveFormat.TAR_GZ.value))]
+
+        # Store period type to easily define how to work with data
+        # as historical and recent data is stored differently
+        file_index_period[DWDMetaColumns.PERIOD_TYPE.value] = period_type
+
+        # Require datetime of file for filtering
+        file_index_period[DWDMetaColumns.DATETIME.value] = file_index_period[DWDMetaColumns.FILENAME.value].\
+            apply(lambda x: re.findall(radolan_dt_regex, x)[0]).\
+            apply(lambda x: pd.to_datetime(x, format=radolan_dt_format))
+
+        file_index = file_index.append(file_index_period)
```

Comment:
Keep in mind: Appending to a DataFrame could yield to a performance issue if the dataFrame gets huge. I think this dataframe is small so doing this in another way will have no effect. Check this out.